packages/components/tsconfig.json

```
@@ -27,6 +27,7 @@
 		"src/base-field/**/*",
 		"src/button/**/*",
 		"src/card/**/*",
+		"src/confirm/**/*",
```

I believe you forgot to update this path after renaming the component:

```suggestion
		"src/confirm-dialog/**/*",
```

This is the reason why we were getting some weird TypeScript behaviour on the component. Rebasing this PR on latest trunk and fixing this path should reveal more TypeScript issues and put us on the right track!

---

Oh, I missed this comment 🤦🏻 GH's UI gets really bad to navigate when there's a lot of comments in a PR.